src/v2/Apps/Fair/FairApp.tsx

```diff
@@ -24,104 +17,86 @@ import {
 } from "@artsy/cohesion"
 import { HttpError } from "found"
 import { userIsAdmin } from "v2/Utils/user"
-import { RouterLink } from "v2/System/Router/RouterLink"
+import { FairHeaderImageFragmentContainer } from "./Components/FairHeader/FairHeaderImage"
 
 interface FairAppProps {
   fair: FairApp_fair
 }
 
 const FairApp: React.FC<FairAppProps> = ({ children, fair }) => {
-  const { user } = useSystemContext()
   const tracking = useTracking()
   const {
     contextPageOwnerId,
     contextPageOwnerSlug,
     contextPageOwnerType,
   } = useAnalyticsContext()
-
-  const hasArticles = (fair.articlesConnection?.edges?.length ?? 0) > 0
-  const hasCollections = (fair.marketingCollections?.length ?? 0) > 0
-  // @ts-expect-error STRICT_NULL_CHECK
-  const artworkCount = fair.counts.artworks
-
-  const clickedArtworksTabTrackingData: ClickedNavigationTab = {
-    action: ActionType.clickedNavigationTab,
-    context_module: ContextModule.exhibitorsTab,
-    context_page_owner_id: contextPageOwnerId,
-    context_page_owner_slug: contextPageOwnerSlug,
-    // @ts-expect-error STRICT_NULL_CHECK
-    context_page_owner_type: contextPageOwnerType,
-    destination_path: `${fair.href}/artworks`,
-    subject: "Artworks",
+  const [lastClickedTab, setLastClickedTab] = useState(ContextModule.fairInfo)
```

Comment:
The current code works and everything is ok. But it seems to me it would be preferable to keep the last taboo using `useRef`

```javascript
const lastClickedTab = useRef(ContextModule.fairInfo)
...
lastClickedTab.current = newValue
```

Reply:
Yeah — in general use state for things that require re-renders. Use refs for all other values.